common/src/main/java/com/microsoft/identity/common/internal/util/StringUtil.java

```diff
@@ -153,4 +154,25 @@ public static int compareSemanticVersion(final String thisVersion, final String
 
         return 0;
     }
+
+    /**
+     * Counts the number of occurrences of one String in another.
+     *
+     * @param str
+     * @param subString
+     * @return int
+     */
+    public static int countMatches(@NonNull String str, @Nullable String subString) {
+        int count = 0;
+        if (StringUtil.isEmpty(str) || StringUtil.isEmpty(subString)) {
+            return count;
+        }
+        for(int i = 0; i < (str.length() - subString.length()); i++) {
+            if(String.copyValueOf(str.toCharArray(), i, subString.length()).equalsIgnoreCase(subString)) {
```

Comment:
Why are we converting string to char array and then using String.copyValueOf? The string `substring` method can achieve the same result with less work I believe. Is there a reason you chose to do it this way?